operator/pkg/controllers/etcd/etcd.go

```diff
@@ -36,14 +36,14 @@ const (
 )
 
 type Controller struct {
-	kubeClient   *kubeprovider.Client
-	certificates *certificates.Provider
+	kubeClient *kubeprovider.Client
+	keypairs   *keypairs.Provider
```

Comment:
Name suggestion: `pkiProvider pki.Provider`

Reply:
`keypairs` pkg is actually a layer over PKI pkg. It provides logic for checking the API server, if secret exists, if not generate certs/keys.